### DISCUSSION TITLE
Testing 15 games with MoltenVK support

### DISCUSSION THREAD

**Discussion by billhollings** [KhronosGroup/MoltenVK#1829] [General]
Many potential game devs and users interested in MoltenVK are curious about what games are using MoltenVK, and what kind of performance MoltenVK supports.

I recently came across this really well-run evaluation on YouTube...

[Testing 15 games with MoltenVK support](https://youtu.be/xDGQcjqpYqI)

If anyone has any other MoltenVK game evaluation results or links, please post them here too.

**Comment by rcaridade145** [comment]
Retroarch has chosen MoltenVK on MacOS:

https://www.libretro.com/index.php/retroarch-1-15-0-release/

> We have made the decision to default the Metal Universal build to the Vulkan video driver. It is currently running faster than the native Metal video driver, has HDR support, and it allows several Vulkan-accelerated cores to run on macOS.

https://github.com/cemu-project/Cemu/pull/693

Cemu will use Async shader compilation once the next version of MoltenVK is released.

**Reply by billhollings** [reply]
Both announcements are great news! Thanks for posting here!

**Comment by K0bin** [comment]
Dark Souls Remastered is in that original video but it's completely broken as soon as you reach Blight Town. The game relies on VK_EXT_robustness2 buffer robustness, otherwise it renders garbage geometry.

**Reply by billhollings** [reply]
Thanks for the feedback. That is disappointing to hear. 

Unfortunately, Metal does not have native robust buffer and image access support. We could potentially do things like adding bounds checking into the converted MSL shader code, but there are concerns that it might cause a noticeable slow-down in shader performance. But perhaps that tradeoff would be useful for some apps.

**Reply by rcaridade145** [reply]
https://github.com/doitsujin/dxvk/issues/405

**Reply by rcaridade145** [reply]
Some info with performance stats :

https://github.com/gpuweb/gpuweb/issues/35
https://github.com/gpuweb/gpuweb/issues/1202

**Reply by billhollings** [reply]
@rcaridade145 Thanks for the research!
